File: java/07 - Demo 3 - Refactoring/src/test/java/decomposeconditional/TireChangeTest.java
package decomposeconditional;

import static org.junit.Assert.assertEquals;

import java.util.Calendar;

import org.junit.Before;
import org.junit.Test;

public class TireChangeTest {

	private static final int SUMMERRATE = 20;
	private static final int WINTERSERVICECHARGE = 15;
	private static final int WINTERRATE = 10;
	private static final int QUANTITY = 2;
	private TireChange tireChange;
	
	@Before
	public void setUp() throws Exception {
		tireChange = new TireChange(QUANTITY, WINTERRATE, WINTERSERVICECHARGE, SUMMERRATE);
	}
	
	@Test
	public void tireChargeInWinter() {
		final int expectedWinterChargeResult = QUANTITY * WINTERRATE + WINTERSERVICECHARGE;
		assertEquals(expectedWinterChargeResult, tireChange.charge(Calendar.DECEMBER));
		assertEquals(expectedWinterChargeResult, tireChange.charge(Calendar.JANUARY));
		assertEquals(expectedWinterChargeResult, tireChange.charge(Calendar.FEBRUARY));
	}
	
	@Test
	public void tireChargeNotInWinter() {
		final int expectedSummerCharge = QUANTITY * SUMMERRATE;
		assertEquals(expectedSummerCharge, tireChange.charge(Calendar.MARCH));
		assertEquals(expectedSummerCharge, tireChange.charge(Calendar.APRIL));
		assertEquals(expectedSummerCharge, tireChange.charge(Calendar.MAY));
		assertEquals(expectedSummerCharge, tireChange.charge(Calendar.JUNE));
		assertEquals(expectedSummerCharge, tireChange.charge(Calendar.JULY));
		assertEquals(expectedSummerCharge, tireChange.charge(Calendar.AUGUST));
		assertEquals(expectedSummerCharge, tireChange.charge(Calendar.SEPTEMBER));
		assertEquals(expectedSummerCharge, tireChange.charge(Calendar.OCTOBER));
		assertEquals(expectedSummerCharge, tireChange.charge(Calendar.NOVEMBER));
	}
}

File: csharp/07 - Demo 3 - Refactoring/Refactoring.Tests/DecomposeConditional/TireChangeTest.cs
using Refactoring.DecomposeConditional;
using Xunit;
using FluentAssertions;

namespace Refactoring.Tests.DecomposeConditional
{
    public class TireChangeTest
    {
        private const int SUMMERRATE = 20;
        private const int WINTERSERVICECHARGE = 15;
        private const int WINTERRATE = 10;
        private const int QUANTITY = 2;
        private TireChange _tireChange;

        public TireChangeTest()
        {
            _tireChange = new TireChange(QUANTITY, WINTERRATE, WINTERSERVICECHARGE, SUMMERRATE);
        }

        [Theory]
        [InlineData(1, QUANTITY * WINTERRATE + WINTERSERVICECHARGE)]
        [InlineData(2, QUANTITY * WINTERRATE + WINTERSERVICECHARGE)]
        [InlineData(3, QUANTITY * SUMMERRATE)]
        [InlineData(4, QUANTITY * SUMMERRATE)]
        [InlineData(5, QUANTITY * SUMMERRATE)]
        [InlineData(6, QUANTITY * SUMMERRATE)]
        [InlineData(7, QUANTITY * SUMMERRATE)]
        [InlineData(8, QUANTITY * SUMMERRATE)]
        [InlineData(9, QUANTITY * SUMMERRATE)]
        [InlineData(10, QUANTITY * SUMMERRATE)]
        [InlineData(11, QUANTITY * SUMMERRATE)]
        [InlineData(12, QUANTITY * WINTERRATE + WINTERSERVICECHARGE)]
        public void TireCharge(int month, int expectedResult)
        {
            var charge = _tireChange.Charge(month);
            charge.Should().Be(expectedResult);
        }
        
    }
}